File: miniprogram/utils/route.ts
// 路径规划：后端代理高德驾车/步行接口（坐标系 GCJ-02，与微信 gcj02 定位一致）。
import { get } from './request'

export type RouteMode = 'driving' | 'walking'

export interface RoutePlan {
  mode: RouteMode
  distance_m: number
  duration_s: number
  polyline: Array<[number, number]>
  mocked: boolean
}

export interface LngLat {
  longitude: number
  latitude: number
}

export function planRoute(origin: LngLat, dest: LngLat, mode: RouteMode): Promise<RoutePlan> {
  return get<RoutePlan>('/route/plan', {
    origin_lng: origin.longitude,
    origin_lat: origin.latitude,
    dest_lng: dest.longitude,
    dest_lat: dest.latitude,
    mode
  })
}

export function formatDistance(meters: number): string {
  if (meters < 1000) return `${Math.round(meters)}米`
  return `${(meters / 1000).toFixed(1)}公里`
}

export function formatDuration(seconds: number): string {
  if (seconds < 60) return '1分钟内'
  const minutes = Math.round(seconds / 60)
  if (minutes < 60) return `约${minutes}分钟`
  const hours = Math.floor(minutes / 60)
  const rest = minutes % 60
  return rest === 0 ? `约${hours}小时` : `约${hours}小时${rest}分钟`
}

File: backend/app/schemas/route.py
"""路径规划响应模型。"""

from typing import Literal

from pydantic import BaseModel, Field


class RoutePlanOut(BaseModel):
    mode: Literal["driving", "walking"] = Field(description="出行方式")
    distance_m: float = Field(description="总距离（米）")
    duration_s: int = Field(description="预计耗时（秒）")
    polyline: list[list[float]] = Field(description="路线点串（[经度, 纬度]）")
    mocked: bool = Field(description="是否为无 key 时的 mock 直线降级")
